File: backend/src/main/java/com/recette/allenchang/recipes/dto/requests/UpdateRecipeRequest.java
package com.recette.allenchang.recipes.dto;

import java.util.List;

// UpdateRecipeRequest.java
public record UpdateRecipeRequest(
        String title,
        String description,
        String imageurl,
        List<IngredientRequest> ingredients,
        List<String> steps,
        List<String> tags,
        String difficulty,
        Integer servingSize,
        Integer cookTimeInMinutes) {
}

File: frontend-ios/dto/UpdateRecipeRequest.swift
import Foundation

struct UpdateRecipeRequest: Codable {
    let title: String
    let description: String
    let imageurl: String?
    let ingredients: [IngredientDTO]
    let steps: [String]
    let tags: [String]
    let difficulty: String?
    let servingSize: Int?
    let cookTimeInMinutes: Int?
}